src/compiler/scala/tools/nsc/backend/jvm/PostProcessor.scala

```diff
@@ -13,7 +13,7 @@ import scala.tools.nsc.backend.jvm.opt._
  * Implements late stages of the backend that don't depend on a Global instance, i.e.,
  * optimizations, post-processing and classfile serialization and writing.
  */
-abstract class PostProcessor extends PerRunInit {
+abstract class PostProcessor(statistics: Statistics with BackendStats) extends PerRunInit {
```

Comment:
I realize now that there's a problem here with thread safety, I assume `statistics` is not thread safe. So we'll have to come back to this when we parallelize classfile writing (and possibly other parts of the backend). But this PR doesn't introduce a new problem, before we were just accessing the static `Statistics` object, which was not thread safe either. cc @mkeskells.